File: linux/linux.py
from linux.apt_install.apt import apt_install
from linux.dnf_install.dnf import dnf_install
from linux.pacman_install.pacman import pacman_install


def linux_os():
    import sys

    print("Selected Linux...")
    sys.path.append("./apt_install")
    sys.path.append("./dnf_install")
    sys.path.append("./pacman_install")

    def operating_system():
        print("What is your package manager?")
        print("1. apt")
        print("2. dnf")
        print("3. pacman")

    def get_valid_input():
        while True:
            pm_input = input("Select your operating system (choose 0 to abort): ")

            # Try to convert the input to an integer
            try:
                pm_int = int(pm_input)  # Try to convert to integer
                return pm_int
            except ValueError:
                # If conversion fails, print an error and ask again
                print("Invalid input!")

    operating_system()
    pm = get_valid_input()

    if pm == 0:
        print("Aborting program...")
        exit()

    if pm == 1:
        from apt_install import apt
        apt_install()

    elif pm == 2:
        from dnf_install import dnf
        dnf_install()

    elif pm == 3:
        from pacman_install import pacman
        pacman_install()

    else:
        print("Select again...")
        pm = get_valid_input()

File: linux/linux.rb
def package_manager
  puts "What package manager is on your Linux Distro?"
  puts "1. apt"
  puts "2. dnf"
  puts "3. eopkg"
  puts "4. nix"
  puts "5. pacman"
  puts "6. zypper"
  puts "7. Other"
  pm = gets.chomp.to_i
  pm
end

def checker(pm)
  case pm
  when 1
    puts "Selected apt..."
    Dir.chdir("./apt")
    system("ruby apt.rb")
  when 2
    puts "Selected dnf..."
    Dir.chdir("./dnf")
    system("ruby dnf.rb")
  when 3
    puts "Selected eopkg..."
    Dir.chdir("./eopkg")
    system("ruby eopkg.rb")
  when 4
    puts "Selected nix..."
    Dir.chdir("./nix")
    system("ruby nix.rb")
  when 5
    puts "Selected pacman..."
    Dir.chdir("./pacman")
    system("ruby pacman.rb")
  when 6
    puts "Selected zypper..."
    Dir.chdir("./zypper")
    system("ruby zypper.rb")
  when 7
    puts "These scripts currently do not support other operating systems."
    puts "If you would us to add support, look at CONTRIBUTING.md to see how you can contribute or send us ideas!"
    puts "Have a nice day!"
    exit 0
  else
    puts "Invalid option, please choose again."
    checker(package_manager)
  end
end

system("clear")
puts "Selected Linux..."
pm = package_manager()
checker(pm)
